base/src/main/kotlin/com/freeletics/gradle/util/Extensions.kt

```diff
@@ -6,23 +6,19 @@ import com.android.build.api.dsl.LibraryExtension
 import com.android.build.api.variant.AndroidComponentsExtension
 import com.android.build.api.variant.ApplicationAndroidComponentsExtension
 import com.android.build.api.variant.LibraryAndroidComponentsExtension
+import com.freeletics.gradle.util.KotlinProjectExtensionDelegate.Companion.kotlinProjectExtensionDelegate
 import org.gradle.api.Project
 import org.gradle.api.plugins.JavaPluginExtension
-import org.jetbrains.kotlin.gradle.dsl.KotlinCommonCompilerOptions
 import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
-import org.jetbrains.kotlin.gradle.dsl.KotlinProjectExtension
-import org.jetbrains.kotlin.gradle.tasks.KotlinCompilationTask
 
 public fun Project.java(action: JavaPluginExtension.() -> Unit) {
     extensions.configure(JavaPluginExtension::class.java) {
         it.action()
     }
 }
 
-public fun Project.kotlin(action: KotlinProjectExtension.() -> Unit) {
-    extensions.configure(KotlinProjectExtension::class.java) {
-        it.action()
-    }
+public fun Project.kotlin(action: KotlinProjectExtensionDelegate.() -> Unit) {
+    kotlinProjectExtensionDelegate().action()
 }
```

Comment:
Added a delegate that allows using the new project level `compilerOptions` for Android/JVM (https://kotlinlang.org/docs/whatsnew19.html#project-level-compiler-options-for-kotlin-jvm) and still uses the old task based configuration for multiplatform